### DISCUSSION TITLE
Lab2 co-simulation error: 'mpfr_srcptr' has not been declared

### DISCUSSION THREAD

**Discussion by yukaii2019** [bol-edu/HLS-SOC-Discussions#69] [Q&A] [answered]
大家好：
我在Lab2 執行 co-simulation 時遇到下面的 error，但合成跟C-simulation都可以順利執行，請問有人知道解決方法嗎，謝謝。
![Screen Shot 2023-09-24 at 2 35 40 PM](https://github.com/bol-edu/HLS-SOC-Discussions/assets/56041116/88e91cfa-0ad9-44db-a4eb-a4609c5f3368)

**Comment by BrianEE07** [comment, marked as answer]
嗨，我上網查好像是在系統中有相同的 gmp.h 檔案，優先層級不對所引起的，我的解決方式是在 FIR.h 上面加入以下兩行，cosim 就不會有 error 了，可以參考下圖。

#include <gmp.h>
#define __gmp_const const

![Screenshot 2023-09-24 at 15 34 40](https://github.com/bol-edu/HLS-SOC-Discussions/assets/46804360/7516d9f2-6420-4af1-845e-da01fa3e0067)

可以參考官方這篇的留言：

[Using ap_int.h in my design causes many C/RTL cosimulation errors, mainly in mpfr.h.](https://support.xilinx.com/s/question/0D52E00006xoQLMSA2/using-apinth-in-my-design-causes-many-crtl-cosimulation-errors-mainly-in-mpfrh?language=en_US)

![Screenshot 2023-09-24 at 15 35 01](https://github.com/bol-edu/HLS-SOC-Discussions/assets/46804360/f18f9afc-084a-44e4-bc2c-14c022735757)

**Reply by yukaii2019** [reply]
嗨~ 謝謝同學這個方法有成功!
